### PR TITLE
feat: add ESLint, Prettier & stylelint to ensure code is standardised 

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
- [x] Enable TypeScript parser for ESLint
- [x] Add relevant config files for `ESLint`, `Prettier` & `stylelint`
- [x] Fixed all linting issues by running `npm run lintfix`
- [x] Adds required devDependent packages

### CODE REVIEW

File: components/atoms/Alert.vue [hunk 1]
```diff
@@ -12,123 +15,132 @@
...
-    return {
-      iconComponent
-    }
-  }
-})
+      return {
+        iconComponent,
+      };
+    },
+  });
```
Comment (Atinux) [inline components/atoms/Alert.vue:51]: Actually I am against the `;` 😅

File: components/atoms/Alert.vue [hunk 2]
```diff
@@ -12,123 +15,132 @@
 </template>
 
 <script>
-import { defineComponent, computed } from '@nuxtjs/composition-api'
+  import { defineComponent, computed } from '@nuxtjs/composition-api';
```
Comment (Atinux) [inline components/atoms/Alert.vue:18]: If we can avoid the tabulation here it would be great

**Comment by Atinux** [comment]
Thank you for this PR @vinayakkulkarni <3

Could you check my 2 comments?

File: components/atoms/Alert.vue (hunk 2 above)
Comment (vinayakkulkarni) [reply components/atoms/Alert.vue]: Removed the [`vueIndentStyleAndScript`](https://github.com/nuxtlabs/nuxtlabs.com/pull/41/files#diff-e42472dd65561fe56344281078077fb924cddca92821eadfbc2318cca3f89c9aR12)

File: components/atoms/Alert.vue (hunk 1 above)
Comment (vinayakkulkarni) [reply components/atoms/Alert.vue]: Yep, fixed it :)

File: package.json [hunk 3]
```diff
@@ -6,16 +6,40 @@
     "generate": "nuxt generate",
     "start": "nuxt start",
     "prepare": "husky install",
-    "lint-staged": "lint-staged"
+    "lint-staged": "lint-staged",
+    "lint": "npm run lint:eslint && npm run lint:prettier && npm run lint:css",
+    "lintfix": "npm run lint:eslint:fix && npm run lint:prettier:fix && npm run lint:css:fix",
+    "lint:js": "npm run lint:eslint && npm run lint:prettier",
+    "lint:eslint": "eslint \"{,!(node_modules|.nuxt|dist|static)/**/}*.{js,ts,vue}\" --ignore-path .gitignore",
+    "lint:eslint:fix": "eslint --fix \"{,!(node_modules|.nuxt|dist|static)/**/}*.{js,ts,vue}\" --ignore-path .gitignore",
```
Comment (Atinux) [inline package.json:14]: Can we use a `.eslintignore` instead?

Comment (vinayakkulkarni) [reply package.json:14]: `.gitignore` works perfectly! I've fixed the issue where the googleapi.com url was breaking the `max-len` rule

Comment (vinayakkulkarni) [reply package.json:14]: If we add `.eslintignore`, then we'd also _probably_ have to add `.prettierignore` and would have to keep them in sync for maintainability

Comment (vinayakkulkarni) [reply package.json:14]: FYI, I've fixed the lint issue – https://github.com/nuxtlabs/nuxtlabs.com/runs/3586142463#step:6:1

**Comment by vinayakkulkarni** [comment]
@Atinux: I require some help in fixing the css, If you enable `stylelint` for `.vue` files, there are tons of errors in the CSS `<style>` tags written in the Vue templates:

Screenshot 2021-09-13 at 4 40 39 PM

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -52,19 +52,7 @@ jobs:
...
-          continue_on_error: true
-          git_name: github-actions[bot]
-          git_email: github-actions[bot]@users.noreply.github.com
-          auto_fix: false
-          stylelint: false
-          eslint: true
-          eslint_extensions: js,ts,vue
-          prettier: true
-          prettier_extensions: js,ts,vue
-          neutral_check_on_warning: true
```
Comment (vinayakkulkarni) [inline .github/workflows/ci.yml:67]: Had to remove the third–party plugin cause of [this](https://github.com/nuxtlabs/nuxtlabs.com/runs/3585991775?check_suite_focus=true#step:6:46) :(

but it shows a nice check in [GitHub PRs](https://github.com/vinayakkulkarni/v-offline/pull/580):
Screenshot 2021-09-13 at 4 47 31 PM

File: package.json (hunk 3 above)
Comment (Atinux) [reply package.json:14]: Does it can at least ignore by default what we have in `.gitignore`? (node_modules, dist, .nuxt, etc)

**Comment by Atinux** [comment]
I fixed the stylelint issues, thank you ❤️

**Pull request merged**